tests/test_methods/test_gather.py

```diff
@@ -0,0 +1,22 @@
+
+import pytest
+
+from returns.future import Future, FutureResult
+from returns.io import IO, IOResult
+from returns.methods import gather
+
+
+@pytest.mark.parametrize(('containers', 'expected'), [
+    (
+        (
+            Future.from_value(1),
+            FutureResult.from_value(2),
+            FutureResult.from_failure(None),
+        ),
+        (IO(1), IOResult.from_value(2), FutureResult.from_failure(None)),
+    ),
+    ((), ()),
+])
+def test_gather(containers, expected):
+    """Test partition function."""
+    assert gather(containers) == expected
```

Comment:
This should not be possible. You cannot mix `Future` and `FutureResult` in one operation without an explicit cast.

Reply:
(from mypy's point of view, I mean)

Reply:
It is not supposed to mix IO operations that can and can not fail? Or it is just mypy limitation?